### PR TITLE
Changes the name of the addTimeWindow method to setTimeWindow.

### PR DESCRIPTION
**Pull request by joeldudleyr3**

### CODE REVIEW

File: docs/source/example-code/src/main/kotlin/net/corda/docs/FlowCookbook.kt [hunk 1]
```diff
@@ -284,7 +285,10 @@ object FlowCookbook {
             regTxBuilder.addOutputState(ourOutput)
             regTxBuilder.addCommand(ourCommand)
             regTxBuilder.addAttachment(ourAttachment)
-            regTxBuilder.addTimeWindow(ourTimeWindow)
+
+            // We set the time-window within which the transaction must be notarised using either of:
+            regTxBuilder.timeWindow = ourTimeWindow
+            regTxBuilder.setTimeWindow(serviceHub.clock.instant(), Duration.ofSeconds(30))
```
Comment (kchalkias) [inline docs/source/example-code/src/main/kotlin/net/corda/docs/FlowCookbook.kt]: add a comment on setTimeWindow, so the user can quickly understand the result is defined
as timeWindow = ([time] - [tolerance], [time] + [tolerance]).

File: docs/source/example-code/src/main/java/net/corda/docs/FlowCookbookJava.java [hunk 2]
```diff
@@ -301,7 +302,10 @@ public Void call() throws FlowException {
             regTxBuilder.addOutputState(ourOutput);
             regTxBuilder.addCommand(ourCommand);
             regTxBuilder.addAttachment(ourAttachment);
-            regTxBuilder.addTimeWindow(ourTimeWindow);
+
+            // We set the time-window within which the transaction must be notarised using either of:
+            regTxBuilder.setTimeWindow(ourTimeWindow);
+            regTxBuilder.setTimeWindow(getServiceHub().getClock().instant(), Duration.ofSeconds(30));
```
Comment (kchalkias) [inline docs/source/example-code/src/main/java/net/corda/docs/FlowCookbookJava.java]: again a small comment on setTimeWindow -> ([time] - [tolerance], [time] + [tolerance]).

File: finance/src/main/kotlin/net/corda/flows/TwoPartyDealFlow.kt [hunk 3]
```diff
@@ -185,7 +185,7 @@ object TwoPartyDealFlow {
 
             // And add a request for a time-window: it may be that none of the contracts need this!
             // But it can't hurt to have one.
-            ptx.addTimeWindow(serviceHub.clock.instant(), 30.seconds)
+            ptx.setTimeWindow(serviceHub.clock.instant(), 30.seconds)
```
Comment (kchalkias) [inline finance/src/main/kotlin/net/corda/flows/TwoPartyDealFlow.kt]: maybe we need to change all related comments from _add_ to _set_ or at least explain what add means (i.e. set or update).

File: samples/irs-demo/src/main/kotlin/net/corda/irs/flows/FixingFlow.kt [hunk 4]
```diff
@@ -74,7 +74,7 @@ object FixingFlow {
 
                     // And add a request for a time-window: it may be that none of the contracts need this!
                     // But it can't hurt to have one.
-                    ptx.addTimeWindow(serviceHub.clock.instant(), 30.seconds)
+                    ptx.setTimeWindow(serviceHub.clock.instant(), 30.seconds)
```
Comment (kchalkias) [inline samples/irs-demo/src/main/kotlin/net/corda/irs/flows/FixingFlow.kt]: again explain what _add_ does.
Please check other references that are related to setting a `TimeWindow`, for instance there is a  `fun timeWindow` in TransactionDSLInterpreter.kt (lines 54-57) and change the comments on _add_ to _set or update_.

File: docs/source/example-code/src/main/kotlin/net/corda/docs/FlowCookbook.kt (hunk 1 above)
Comment (joeldudleyr3) [inline docs/source/example-code/src/main/kotlin/net/corda/docs/FlowCookbook.kt]: Thanks, I'll make a note of this - I'm changing the cookbook in a separate PR currently.

File: docs/source/example-code/src/main/java/net/corda/docs/FlowCookbookJava.java (hunk 2 above)
Comment (joeldudleyr3) [inline docs/source/example-code/src/main/java/net/corda/docs/FlowCookbookJava.java]: Thanks

File: samples/irs-demo/src/main/kotlin/net/corda/irs/flows/FixingFlow.kt (hunk 4 above)
Comment (joeldudleyr3) [inline samples/irs-demo/src/main/kotlin/net/corda/irs/flows/FixingFlow.kt]: Thanks

**Pull request merged**